### PR TITLE
[REV-737] Updating build pipeline to publish to GitHub Packages.

### PR DESCRIPTION
**Pull request by tdobbs-alist**

### CODE REVIEW

File: .ci [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+        }
+        stage('Bundle') {
+          steps {
+            withCredentials([usernamePassword(credentialsId: 'ApartmentList-CI', usernameVariable: 'GIT_USERNAME', passwordVariable: 'GIT_TOKEN')]) {
+              container('ruby') {
+                withEnv([
+                  "GIT_USERNAME=${GIT_USERNAME}",
+                  "GIT_TOKEN=${GIT_TOKEN}"
+                ]) {
+                  sh 'apt-get update'
```
Comment (tmizohata-alist) [inline .ci:96]: `apt-get update` doesn't do much in this context, so you can remove it.

File: .ci [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+      parallel {
+        stage('Run RSpec') {
+          steps {
+            container('ruby') {
+              withEnv([
+                "BUNDLE_GITHUB__COM=${BUNDLE_GITHUB__COM}",
+                "BUNDLE_GITHUB__HTTPS=${BUNDLE_GITHUB__HTTPS}"
+              ]) {
+                sh label: 'RSpec', script: 'bundle exec rake'
+              } // withEnv
```
Comment (tmizohata-alist) [inline .ci:124]: I think you can pull `sh()` out of `withEnv()` block because you now use package manager, not github code.

File: .ci [hunk 3]
```diff
@@ -0,0 +1,158 @@
...
+    } // stage - Publish
+  } // stages
+
+  post {
+    success {
+      slackSuccess()
+    } // success
+    failure {
+      slackFailure()
+    } // failure
```
Comment (tmizohata-alist) [inline .ci:154]: we now like this updated way

```
success {
  success('alist-staging')
} // success
aborted {
  aborted('alist-staging')
} // aborted
```

**Review by tmizohata-alist — CHANGES_REQUESTED** [review]
minor nitpicking but thank you!

File: .ci [hunk 4]
```diff
@@ -0,0 +1,158 @@
...
+    CLOUDSDK_CORE_DISABLE_PROMPTS = '1'
+    GIT_COMMIT_SHORT = sh(script: "git rev-parse --short ${GIT_COMMIT}", returnStdout: true).trim()
+    GIT_MESSAGE = sh(script: "git log --format=%B -n 1 ${GIT_COMMIT}", returnStdout: true).trim()
+    GIT_USER = sh(script: "git log -1 --pretty=format:'%ae'", returnStdout: true).trim()
+    GITHUB_URL = "https://github.com"
+    LANG = "en_US.UTF-8"
+    LANGUAGE = "en_US:en"
+    LC_ALL = "en_US.UTF-8"
+    PRODUCTION_DEPLOY="false"
+    SLACK_CHANNEL = "#staging-releases"
```
Comment (tmizohata-alist) [inline .ci:35]: because of the update on `post` this line won't be used as well. you can remove it.

File: .ci (hunk 3 above)
Comment (tmizohata-alist) [reply .ci:154]: and actually this change will require a few more env values:

```
SLACK_SUCCESS_CHANNEL
SLACK_FAILURE_CHANNEL
```

File: .ci [hunk 5]
```diff
@@ -32,7 +32,6 @@ pipeline {
     LANGUAGE = "en_US:en"
     LC_ALL = "en_US.UTF-8"
     PRODUCTION_DEPLOY="false"
-    SLACK_CHANNEL = "#staging-releases"
```
Comment (tmizohata-alist) [inline .ci]: Sorry I made a wrong comment earlier. You do need `SLACK_CHANNEL` if you don't give `SLACK_SUCCESS_CHANNEL` and/or `SLACK_FAILURE_CHANNEL`. And that's why this failed:

https://test.apartmentlist.io/blue/organizations/jenkins/apartmentlist%2Fbulk-processor/detail/PR-34-head/9/pipeline

**Review by tmizohata-alist — CHANGES_REQUESTED** [review]

File: .ci (hunk 3 above)
Comment (tdobbs-alist) [reply .ci]: Any opinions on whether to use alist-staging vs alist-production here? Also relevant, for SLACK_CHANNEL, #staging-releases or ...?

**Review by tmizohata-alist — APPROVED** [review]

**Pull request merged**